cmake/modules/FindDiaSDK.cmake

```diff
@@ -69,6 +69,27 @@ if(NOT DiaSDK_ROOT)
 	endif()
 endif()
 
+# Gather up the include directories listed in the INCLUDE environment variable.
+set(env_includes)
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL "CYGWIN")
+	# INCLUDE is a list of windows style paths.
+	# This needs to be converted to cygwin style paths to feed to CMake.
+	if(CMAKE_CYGPATH AND (NOT "$ENV{INCLUDE}" STREQUAL ""))
+		execute_process(
+			COMMAND "${CMAKE_CYGPATH}" -c $ENV{INCLUDE}
```

Comment:
Perhaps `$ENV{INCLUDE}` should be quoted (`"$ENV{INCLUDE}"`)? It may contain spaces.

Reply:
it needs to be unquoted here so that the separate path components will be separated into separate arguments to cygpath. I believe cmake will implicitly quote spaces that appear as part of a variable expansion. At the very least this code works on my machine which does have spaces in some of the include paths

Reply:
According to [1] and `cygpath --help`, the `-c` option is unwanted here; it expects a 'HANDLE' argument. I think we want
```
COMMAND "${CMAKE_CYGPATH}" "$ENV{INCLUDE}"
```
Also, without the quotes, `cygpath` treats each word as a separate argument writing each result on a separate line.

[1] https://cygwin.com/cygwin-ug-net/cygpath.html

Reply:
I'm not sure how `-c` got there, should be `-u`, although it looks like thats the default anyway. 
Wrapping the env expansion in quotes would suppress the list expansion behavior thats desired here. Spaces inside the quote are not an issue as cmake has implicit quotes around a variable expansion. 
eg. if INCLUDE="a b;c"
`execute_process(COMMAND cygpath $ENV{INCLUDE})` -> `execute_process(COMMAND cygpath "a b" "c")`
where as 
`execute_process(COMMAND cygpath "$ENV{INCLUDE}")` -> `execute_process(COMMAND cygpath "a b;c")`

Reply:
I think we want the latter. On Windows, the path separator is `;` which should be visible in the argument to `cygpath` so it can be replaced by `:`, the unix path separator.

Reply:
yes, the idea was to intentionally use the fact that the windows path separator is the same as the cmake path separator. This means that each element in the path list is processed as a separate argument so that it is translated properly.

```
$ cygpath -u "c:\\foo" "c:\\bar"
/cygdrive/c/foo
/cygdrive/c/bar

$ cygpath -u "c:\\foo;c:\\bar"
/cygdrive/c/foo;c:/bar
```

Reply:
Ok, I get it now. Perhaps some comments explaining this would help.